src/averageAges.js

```diff
@@ -55,7 +77,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const checkMother = motherName => people.find(
+    person => person.name === motherName
+  );
+  const manChecking = person => person.sex === 'm';
+
+  const children = onlyWithSon === true
+    ? people.filter(person => checkMother(person.mother) && manChecking(person))
```

Comment:
You can move a ternary operator inside the find function to not repeat it twice

Reply:
How could i do it?